### ISSUE TITLE
Upload via share link from the gallery view

### ISSUE THREAD

**Issue by jancborchardt** [owncloud/gallery#225]
Similar to https://github.com/owncloud/galleryplus/issues/25 (upload straight from the app) but in the share link view.

Use-case: I often share picture folders with friends who also made photos from the same event. Since I want them to add their photos too, I send them the normal share link because there the upload button shows. But it’s the lame list view – I’d much rather have them see the pictures in the nice photo layout, but also be able to add their photos.

@oparoz I assume this is blocked by core just like #25? What is necessary to make it work? Can you maybe even submit a pull request for it?
### TODO
- [x] Upload straight from the app #25 
- [x] Add missing options to the share dialogue https://github.com/owncloud/gallery/issues/565
- [ ] Fix owncloud/core#23066 via PR #652 
##

**Comment by oparoz** [comment]
The problem is that the upload mechanism is a dated hack, made of strings and which is not in the public space, so using it as-is a recipe for a maintenance nightmare imho. It also violates some oC rules which stipulate we should not create inter-app dependencies.

**Comment by jancborchardt** [comment]
Ok, then we need to reimplement the upload in Gallery to fix this? Or can we safely assume Files is always the base? At least Gallery does not make any sense without Files as base. cc @DeepDiver1975

**Comment by oparoz** [comment]
That's what we need:
https://github.com/owncloud/core/issues/17610
but it's in the backlog.

The whole IE compatibility requirement also makes it difficult to introduce new features in official apps via HTML5 classes.

I'd say we should keep this on standby for when Gallery becomes official, because like you said Files is a crucial part of `core` and it wouldn't be a big deal for an official app to use its scripts if this requirement rises to the top of the queue.

**Comment by oparoz** [comment]
@karlitschek - Can we make this an HTML5, modern browser only feature? I think it's a nice-to-have feature, not a hard requirement, just like upload works differently in Files, depending on the browser used.

**Comment by karlitschek** [comment]
@MTRichards what do you think?

**Comment by MTRichards** [comment]
Ok for me if the coolest new stuff only works with a new browser, as long as the IE8 view is not totally mashed, just has to be performed in a different way.

**Comment by oparoz** [comment]
OK, thanks. It's not for right now anyway, but if I stumble upon a simple way to easily upload a bunch of images to the current folder without loading a large library, it's probably going to be for modern browsers only. I'll know more when I start looking.

**Comment by karlitschek** [comment]
good discussion guys :-)

**Comment by jancborchardt** [comment]
Yeah – if IE is a blocker for such a major feature, I’d say it’s fine to have it just for the modern browsers. If you only use IE, you will miss nothing.

**Comment by oparoz** [comment]
@jancborchardt - The first step for this is #25 which will include the necessary libraries and methods to safely upload pictures while avoiding overwriting existing content.

**Comment by jancborchardt** [comment]
Maybe this is something we could advertise via a blog post »Where you can help improve ownCloud« @jospoortvliet? Similar to contributing to the Calendar & Contacts rework?

**Comment by oparoz** [comment]
I was also thinking of using something like indiegogo to fund the development of all the features needing a sponsor (like uploading) because I usually have a pretty good idea of how they should be implemented, but that would require some marketing in order to make ownCloud users aware of it.

**Comment by jancborchardt** [comment]
That’s kinda what us being on Bountysource is for. ;) Of course the advertising / marketing to make people aware of that is missing there too.

**Comment by oparoz** [comment]
The problem is that they've dropped support for campaigns and voluntary donations cover 1 lunch, not a week of development, so yes, letting users know is key.

**Comment by oparoz** [comment]
This can now be worked on and should be fairly easy to do since uploading is in place.

**Comment by oparoz** [comment]
This issue was moved to nextcloud/gallery#54
